### PR TITLE
Feature: migration rules for the 'isSelfAnActiveMember' in the Conversation model

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Add:
- migration rule for the isSelfAnActiveMember property,
- getRoles() method to the Conversation model,
- test to check a new method getRoles(),
- remove isSelfAnActiveMembe,

### CODE REVIEW

**Comment by zenkins** [comment]
Can one of the admins verify this patch?

File: Source/Model/Conversation/ZMConversation+Participants.swift [hunk 1]
```diff
@@ -156,4 +156,11 @@ extension ZMConversation {
         }
     }
     
+    // MARK: - Conversation roles
+    
+    /// List of roles for the conversation whether it's linked with a team or not
+    @objc
+    public func getRoles() -> Set<Role> {
+        return (self.team == nil) ? self.nonTeamRoles : self.team!.roles
```
Comment (billypchan) [inline Source/Model/Conversation/ZMConversation+Participants.swift]:
```suggestion
        if let team = team {
           return team.roles
        }
        return nonTeamRoles
```

File: Source/Model/Conversation/ZMConversation+Patches.swift [hunk 2]
```diff
@@ -37,4 +38,39 @@ extension ZMConversation {
...
+                    participantRoleForSelfUser = getAParticipantRoleForANonTeamMember(in: moc, adminRole: adminRole, user: selfUser, conversation: conversation)
+                }
+                conversation.participantRoles.insert(participantRoleForSelfUser)
+            }
+        }
+    }
+    
+    static private func getAParticipantRoleForATeamMember(in moc: NSManagedObjectContext, adminRole: Role?, user: ZMUser, conversation: ZMConversation, team: Team) -> ParticipantRole {
+        let participantRoleForUser = ParticipantRole.create(managedObjectContext: moc, user: user, conversation: conversation)
+        participantRoleForUser.role = (adminRole == nil) ? Role.create(managedObjectContext: moc, name: defaultAdminRoleName, team: team) : adminRole!
```
Comment (billypchan) [inline Source/Model/Conversation/ZMConversation+Patches.swift]: Should we merge this method with `getAParticipantRoleForANonTeamMember`, with a optional paramter `team`?

File: Source/Model/Conversation/ZMConversation+Participants.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation+Participants.swift]: Done

File: Source/Model/Conversation/ZMConversation+Participants.swift [hunk 3]
```diff
@@ -36,8 +42,12 @@ extension ZMConversation {
...
     public class func keyPathsForValuesAffectingActiveParticipants() -> Set<String> {
-        return Set([ZMConversationIsSelfAnActiveMemberKey,
-                    ZMConversationParticipantRolesKey])
+        return Set([ZMConversationParticipantRolesKey])
+    }
+    
+    @objc
+    public class func keyPathsForValuesAffectingIsSelfAnActiveMember() -> Set<String> {
+        return Set([ZMConversationParticipantRolesKey])
     }
```
Comment (billypchan) [inline Source/Model/Conversation/ZMConversation+Participants.swift]: It is same as `keyPathsForValuesAffectingActiveParticipants`, should this returns a key for `isSelfAnActiveMember`?

File: Source/Model/Conversation/ZMConversation+Participants.swift [hunk 4]
```diff
@@ -119,7 +129,7 @@ extension ZMConversation {
...
     
     @objc
-    func minus(userSet: Set<ZMUser>, isFromLocal: Bool) {
+    public func minus(userSet: Set<ZMUser>, isFromLocal: Bool) {
```
Comment (billypchan) [inline Source/Model/Conversation/ZMConversation+Participants.swift]: I will refactor this with variadic parameter (`users: ZMUser...`) to reduce duplicated interfaces.

File: Source/Model/Conversation/ZMConversation+Patches.swift (hunk 2 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation+Patches.swift]: Fixed

File: Source/Model/Conversation/ZMConversation.m [hunk 5]
```diff
@@ -345,7 +347,8 @@ - (NSSet *)ignoredKeys;
             HasReadReceiptsEnabledKey,
             ZMConversationLegalHoldStatusKey,
             ZMConversationNeedsToVerifyLegalHoldKey,
-            ZMConversationLabelsKey
+            ZMConversationLabelsKey,
+            @"isSelfAnActiveMember"
```
Comment (billypchan) [inline Source/Model/Conversation/ZMConversation.m]: We should create a constant sting for this?

**Pull request merged**